test/storage/sql_table_concurrent_test.cpp

```diff
@@ -0,0 +1,379 @@
+    std::vector<storage::TupleSlot> tuples;
+
+    // Setup table
+    transaction::TransactionContext *init_txn = txn_manager_.BeginTransaction();
+
+    auto row_pair = table.InitializerForProjectedRow(*versioned_col_oids[!schema_version_], schema_version_);
+    auto pri = new storage::ProjectedRowInitializer(std::get<0>(row_pair));
+    auto pr_map = new storage::ProjectionMap(std::get<1>(row_pair));
+
+    for (uint32_t i = 0; i < txns_per_thread * num_threads; i++) {
+      byte *buffer = common::AllocationUtil::AllocateAligned(pri->ProjectedRowSize());
+      auto pr = pri->InitializeRow(buffer);
+
+      PopulateProjectedRow(schema_version_, i, pr, pr_map);
+
+      tuples.emplace_back(table.Insert(init_txn, *pr, schema_version_));
+      pr = nullptr;
+      delete[] buffer;
+    }
+    txn_manager_.Commit(init_txn, TestCallbacks::EmptyCallback, nullptr);
+
+    delete init_txn;
+
+    delete pri;
+    delete pr_map;
+
+    // Begin concurrent section
+    auto workload = [&](uint32_t id) {
+      for (uint32_t t = 0; t < txns_per_thread; t++) {
+        storage::layout_version_t working_version = schema_version_;
+        transaction::TransactionContext *txn;
+        if (id == 0) {
+          if (t >= 8) break;  // No more schema updates
+          txn = txn_manager_.BeginTransaction();
+          // LOG_INFO("  Adding schema version {}", (!working_version)+1);
+          catalog::Schema schema(GenerateColumnsVector(working_version + 1), working_version + 1);
+          table.UpdateSchema(schema);
+          // Update schema
+        } else {
+          txn = txn_manager_.BeginTransaction();
+          // LOG_INFO("    Thread {} in version {} with transaction {}", id, !working_version, t);
+          auto row_pair = table.InitializerForProjectedRow(*versioned_col_oids[!working_version], working_version);
+          auto pri = new storage::ProjectedRowInitializer(std::get<0>(row_pair));
+          auto pr_map = new storage::ProjectionMap(std::get<1>(row_pair));
+          int32_t base_val = id * txns_per_thread + t;
+          byte *buffer = common::AllocationUtil::AllocateAligned(pri->ProjectedRowSize());
+          auto pr = pri->InitializeRow(buffer);
+
+          table.Select(txn, tuples[base_val], pr, *pr_map, working_version);
+          ValidateTuple(pr, pr_map, working_version, base_val);
+
+          pr = nullptr;
+          delete[] buffer;
+          delete pri;
+          delete pr_map;
+        }
+        txn_manager_.Commit(txn, TestCallbacks::EmptyCallback, nullptr);
+        if (id == 0) schema_version_++;
+        delete txn;
+      }
+    };
+
+    MultiThreadTestUtil::RunThreadsUntilFinish(&thread_pool, num_threads, workload);
+    for (auto &version : versioned_col_oids) delete version;
+    versioned_col_oids.clear();
+    // End concurrent section
+    // delete init_txn;
+  }
+}
+
+// NOLINTNEXTLINE
+TEST_F(SqlTableConcurrentTests, ConcurrentQueriesWithSchemaChange) {
+  // const uint32_t num_iterations = 5;
```

Comment:
This file has many commented out lines, even including two tests, which does not look good.

Reply:
You are correct.  These are placeholders for two more tests that we need to write.